packages/amplify-migration-codegen-e2e/src/assertions.ts

```diff
@@ -5,50 +5,108 @@ import {
+  const { UserPoolId: gen1UserPoolId } = Object.keys(gen1Meta.auth).map((key) => gen1Meta.auth[key])[0].output;
   const cloudUserPool = await getUserPool(gen1UserPoolId, gen1Region);
   expect(cloudUserPool.UserPool).toBeDefined();
+  return { gen1UserPoolId };
+}
 
+async function assertFunction(gen1Meta: $TSAny, gen1Region: string) {
+  const { Arn: gen1FunctionArn, Name: gen1FunctionName } = Object.keys(gen1Meta.function).map((key) => gen1Meta.function[key])[0].output;
   expect(gen1FunctionArn).toBeDefined();
   expect(gen1FunctionName).toBeDefined();
   const cloudFunction = await getFunction(gen1FunctionName, gen1Region);
   expect(cloudFunction.Configuration?.FunctionArn).toEqual(gen1FunctionArn);
+  return { gen1FunctionName };
+}
 
+async function assertStorage(gen1Meta: $TSAny, gen1Region: string) {
+  const { BucketName: gen1BucketName } = Object.keys(gen1Meta.storage).map((key) => gen1Meta.storage[key])[0].output;
   expect(gen1BucketName).toBeDefined();
   const bucketExists = await checkIfBucketExists(gen1BucketName, gen1Region);
   expect(bucketExists).toMatchObject({});
+  return { gen1BucketName };
+}
+
+async function assertAPI(gen1Meta: $TSAny, gen1Region: string) {
+  const {
+    GraphQLAPIIdOutput: gen1GraphqlApiId,
+    GraphQLAPIEndpointOutput,
+    GraphQLAPIKeyOutput,
+  } = Object.keys(gen1Meta.api).map((key) => gen1Meta.api[key])[0].output;
+  const { graphqlApi } = await getAppSyncApi(gen1GraphqlApiId, gen1Region);
 
   expect(gen1GraphqlApiId).toBeDefined();
   expect(GraphQLAPIEndpointOutput).toBeDefined();
   expect(GraphQLAPIKeyOutput).toBeDefined();
 
   expect(graphqlApi).toBeDefined();
   expect(graphqlApi?.apiId).toEqual(gen1GraphqlApiId);
+  return { gen1GraphqlApiId };
+}
+
+export async function assertDefaultGen1Setup(projRoot: string) {
+  const { gen1Meta, gen1Region } = await getMeta(projRoot);
+  const { gen1UserPoolId } = await assertUserPool(gen1Meta, gen1Region);
+  const { gen1FunctionName } = await assertFunction(gen1Meta, gen1Region);
+  const { gen1BucketName } = await assertStorage(gen1Meta, gen1Region);
+  const { gen1GraphqlApiId } = await assertAPI(gen1Meta, gen1Region);
+
   return { gen1UserPoolId, gen1FunctionName, gen1BucketName, gen1GraphqlApiId, gen1Region };
 }
 
+export async function assertAuthWithMaxOptionsGen1Setup(projRoot: string) {
+  const { gen1Meta, gen1Region } = await getMeta(projRoot);
+  const { gen1UserPoolId } = await assertUserPool(gen1Meta, gen1Region);
+  const { gen1FunctionName } = await assertFunction(gen1Meta, gen1Region);
+
+  const {
+    UserPoolId: userPoolId,
+    AppClientIDWeb: appClientIdWeb,
+    AppClientID: appClientId,
+  } = Object.keys(gen1Meta.auth).map((key) => gen1Meta.auth[key])[0].output;
+  const clientIds = [appClientIdWeb, appClientId];
+  const clients = await getUserPoolClients(userPoolId, clientIds, gen1Region);
+  expect(clients).toHaveLength(2);
+
+  const { userPoolGroups } = gen1Meta.auth;
```

Comment:
nit: pull it to `assertUserPoolGroups()` if you think it is useful later (reusability).

Reply:
Done [dbf44b4](https://github.com/aws-amplify/amplify-cli/pull/13978/commits/dbf44b417f56692be3608d684dc8f31f985b6df8)